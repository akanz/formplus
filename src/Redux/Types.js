export const CHANGE_SEARCH_FIELD = "CHANGE_SEARCH_FIELD"
export const FETCH_TEMPLATE_DETAILS = "FETCH_TEMPLATE_DETAILS"
export const FETCH_TEMPLATE_REQUEST = "FETCH_TEMPLATE_REQUEST"
export const FETCH_TEMPLATE_SUCCESS = "FETCH_TEMPLATE_SUCCESS"
export const FETCH_TEMPLATE_ERROR = "FETCH_TEMPLATE_ERROR"
export const CHANGE_CATEGORY = 'CHANGE_CATEGORY'
export const CHANGE_ORDER_FIELD = "CHANGE_ORDER_FIELD"
export const CHANGE_DATE_FIELD = 'CHANGE_DATE_FIELD'
export const CHANGE_SELECT_VALUE = 'CHANGE_SELECT_VALUE'

