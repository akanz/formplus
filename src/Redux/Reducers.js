import {
  CHANGE_SEARCH_FIELD,
  FETCH_TEMPLATE_DETAILS,
  FETCH_TEMPLATE_SUCCESS,
  FETCH_TEMPLATE_ERROR,
  FETCH_TEMPLATE_REQUEST,
  CHANGE_CATEGORY,
  CHANGE_ORDER_FIELD,
  CHANGE_DATE_FIELD,
} from "./Types";

export const initialstate = {
  searchfield: "",
  loading: false,
  error: "",
  categoryfield: "all",
  orderfield: "default",
  datefield: "default",
  Data: [],
};

const reducer = (state = initialstate, action) => {
  switch (action.type) {
    case CHANGE_SEARCH_FIELD:
      return {
        ...state,
        searchfield: action.payload,
      };
    case FETCH_TEMPLATE_DETAILS:
      return {
        ...state,
        Data: action.payload,
      };
    case FETCH_TEMPLATE_REQUEST:
      return {
        ...state,
        loading: true,
      };
    case FETCH_TEMPLATE_SUCCESS:
      return {
        ...state,
        loading: false,
        Data: action.payload,
      };
    case FETCH_TEMPLATE_ERROR:
      return {
        ...state,
        loading: false,
        error: action.payload,
      };
    case CHANGE_CATEGORY:
      return {
        ...state,
        categoryfield: action.payload,
      };
    case CHANGE_ORDER_FIELD:
      return {
        ...state,
        orderfield: action.payload,
      };
    case CHANGE_DATE_FIELD:
      return {
        ...state,
        datefield: action.payload,
      };
    default:
      return state;
  }
};
export default reducer;
