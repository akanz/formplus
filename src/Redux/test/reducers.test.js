import * as types from '../Types'
import reducer from '../Reducers'
import {initialstate} from '../Reducers'


describe('All reducers', () => {

  it('should return the initial state', () => {
    expect(reducer(undefined, {})).toMatchSnapshot()
  })

  it('should handle CHANGE_CATEGORY',()=> {
    const category = reducer(initialstate, { type: types.CHANGE_CATEGORY });
      expect(category).toMatchSnapshot()
  })
  it('should handle CHANGE_ORDER_FIELD',()=> {
    const order = reducer(initialstate, { type:types.CHANGE_ORDER_FIELD });
    expect(order).toMatchSnapshot()
  })
  it('should handle CHANGE_SEARCH_FIELD', ()=> {
    const search = reducer(initialstate, { type:types.CHANGE_SEARCH_FIELD });
    expect(search).toMatchSnapshot()
  })
  it('should handle CHANGE_DATE_FIELD', ()=> {
    const date = reducer(initialstate, { type:types.CHANGE_DATE_FIELD });
    expect(date).toMatchSnapshot()
  })
 
  it('should handle FETCH_REQUEST', ()=> {
    const fetch = reducer(initialstate, { type:types.FETCH_TEMPLATE_REQUEST });
    expect(fetch).toMatchSnapshot()
  })
  it('should handle FETCH_TEMPLATE_SUCCESS', ()=> {
    const fetch = reducer(initialstate, { type:types.FETCH_TEMPLATE_SUCCESS });
    expect(fetch).toMatchSnapshot()
  })
  it('should handle FETCH_ERROR', ()=> {
    const fetch = reducer(initialstate, { type:types.FETCH_TEMPLATE_ERROR });
    expect(fetch).toMatchSnapshot()
  })

})