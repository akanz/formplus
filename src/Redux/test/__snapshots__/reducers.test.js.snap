// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`All reducers should handle CHANGE_CATEGORY 1`] = `
Object {
  "Data": Array [],
  "categoryfield": undefined,
  "datefield": "default",
  "error": "",
  "loading": false,
  "orderfield": "default",
  "searchfield": "",
}
`;

exports[`All reducers should handle CHANGE_DATE_FIELD 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": undefined,
  "error": "",
  "loading": false,
  "orderfield": "default",
  "searchfield": "",
}
`;

exports[`All reducers should handle CHANGE_ORDER_FIELD 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": "default",
  "error": "",
  "loading": false,
  "orderfield": undefined,
  "searchfield": "",
}
`;

exports[`All reducers should handle CHANGE_SEARCH_FIELD 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": "default",
  "error": "",
  "loading": false,
  "orderfield": "default",
  "searchfield": undefined,
}
`;

exports[`All reducers should handle FETCH_ERROR 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": "default",
  "error": undefined,
  "loading": false,
  "orderfield": "default",
  "searchfield": "",
}
`;

exports[`All reducers should handle FETCH_REQUEST 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": "default",
  "error": "",
  "loading": true,
  "orderfield": "default",
  "searchfield": "",
}
`;

exports[`All reducers should handle FETCH_TEMPLATE_SUCCESS 1`] = `
Object {
  "Data": undefined,
  "categoryfield": "all",
  "datefield": "default",
  "error": "",
  "loading": false,
  "orderfield": "default",
  "searchfield": "",
}
`;

exports[`All reducers should return the initial state 1`] = `
Object {
  "Data": Array [],
  "categoryfield": "all",
  "datefield": "default",
  "error": "",
  "loading": false,
  "orderfield": "default",
  "searchfield": "",
}
`;
